clippy_lints/src/single_char_push_str.rs

```diff
@@ -0,0 +1,62 @@
+use crate::utils::{match_def_path, paths, snippet_with_applicability, span_lint_and_sugg};
+use if_chain::if_chain;
+use rustc_ast::ast::LitKind;
+use rustc_errors::Applicability;
+use rustc_hir::{Expr, ExprKind};
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+
+declare_clippy_lint! {
+    /// **What it does:** Warns when using push_str with a single-character string literal,
+    /// and push with a char would work fine.
+    ///
+    /// **Why is this bad?** This is in all probability not the intended outcome. At
+    /// the least it hurts readability of the code.
+    ///
+    /// **Known problems:** None
+    ///
+    /// **Example:**
+    /// ```
+    /// let mut string = String::new();
+    /// string.push_str("R");
+    /// ```
+    /// Could be written as
+    /// ```
+    /// let mut string = String::new();
+    /// string.push('R');
+    /// ```
+    pub SINGLE_CHAR_PUSH_STR,
+    style,
```

Comment:
I think this is correct. The issue mentioned better performance, but this does not seem to be true as the generated assembler is exactly the same. See [here](https://godbolt.org/z/3vja7c) and [here](https://godbolt.org/z/zjTcW1).

Reply:
It looks like rustc is optimizing the `push_str("R")` to a `push('R')` so it's not a performance difference.

Reply:
Sorry if I was not precise enough, I meant that Clippy's suggestion would not improve the performance.